### PR TITLE
start work on value api

### PR DESCRIPTION
**Pull request by havardpe**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@arnej27959 please review

### CODE REVIEW

**Review by arnej27959 — APPROVED** [review]
👍

File: eval/src/vespa/eval/eval/simple_value.h [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+ * is intended as a reference implementation that can also be used to
+ * test the correctness of tensor operations as they are moved away
+ * from the implementation of individual tensor classes.
+ **/
+class SimpleValue : public NewValue, public NewValue::Index
+{
+private:
+    using Addr = std::vector<vespalib::string>;
+    ValueType _type;
+    size_t _num_mapped;
```
Comment (geirst) [inline eval/src/vespa/eval/eval/simple_value.h:148]: Consider rename __num_mapped_ -> __num_mapped_dims_ to be more explicit on what it represents. This also applies for same variable name used in ValueBuilderFactory.

File: eval/src/vespa/eval/eval/simple_value.h [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+ * passed since it is exposed in the value api. The expected number of
+ * subspaces is also passed since it enables the builder to pre-size
+ * internal structures appropriately. Note that since we are not able
+ * to have virtual templated functions we need to cast the created
+ * builder. With interoperability between all values.
+ **/
+struct ValueBuilderFactory {
+    template <typename T>
+    std::unique_ptr<ValueBuilder<T>> create_value_builder(const ValueType &type,
+            size_t num_mapped_in, size_t subspace_size_in, size_t expect_subspaces) const
```
Comment (geirst) [inline eval/src/vespa/eval/eval/simple_value.h:116]: Consider rename _expect_subspaces_ -> _expected_subspaces_, as the current name sounds like it should be a bool.

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**